torch/csrc/jit/serialization/export.cpp

```diff
@@ -1271,8 +1271,35 @@ void GraphEncoder::EncodeTensor(
       location->set_value(tensorName);
       tensor_proto->set_data_location(onnx::TensorProto_DataLocation_EXTERNAL);
     } else {
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
       tensor_proto->set_raw_data(std::string(
           static_cast<char*>(t.data_ptr()), t.element_size() * t.numel()));
+#elif __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
```

Comment:
`EncodeTensor` is a big function as is. Moving this code to a utility would improve readability and reusability